### PR TITLE
Update readme local build instructions

### LINKED ISSUE #21198: Update README.md local build instructions
### What is this about?

The local build instructions for contributors in the README.md file needs to be brought up-to-date. I also have ideas for fixes and improvements:

- Add step for installing dependencies.
- Add step for running local build in the browser.
- Add instructions for resolving [Husky issue](https://typicode.github.io/husky/troubleshooting.html#command-not-found) when making commits in VS Code.
- Fix outdated link for Infura environment variables.
- Fix typo in `.metamaskrc.dist`.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### References

- Closed by https://github.com/MetaMask/metamask-extension/pull/20997

### PR DESCRIPTION
**Pull request by MajorLift**
### Improvements
- Added step for installing deps before running `yarn dist`. 
  - Instructions should be as clear as possible with no implicit steps.
- Added final step of running the local build in the browser.
  - Useful as additional check to confirm that build runs without issues.
  - Provides rewarding visual feedback for first-time contributors.
- Added instructions for resolving [Husky issue](https://typicode.github.io/husky/troubleshooting.html#command-not-found) when making commits in VS Code.

### Fixes
- Infura no longer uses a project ID, but an [API key](https://docs.infura.io/networks/ethereum/how-to/secure-a-project/project-id).
- Fixed typo in `.metamaskrc.dist`.

### Notes
- Renaming the `INFURA_PROJECT_ID` environment variable to `INFURA_API_KEY` might be helpful for reducing confusion down the line.
- Issues with Redux devtools set-up instructions: 
  - Obsolete use of `yarn global` command needs to be updated. 
  - It's also unclear if `remotedev-server` is still needed as redux devtools seems to be functional without running `yarn devtools:redux` build script. 
  - Will ask around and open separate issue if necessary.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by MajorLift** [comment]
I have read the CLA Document and I hereby sign the CLA

File: README.md [hunk 1]
```diff
@@ -49,7 +56,8 @@ To start the [React DevTools](https://github.com/facebook/react-devtools), run `
 
 To start the [Redux DevTools Extension](https://github.com/reduxjs/redux-devtools/tree/main/extension):
 
-- Install the package `remotedev-server` globally (e.g. `yarn global add remotedev-server`)
+- Install the package `remotedev-server` and `sqlite3` (e.g. `yarn add remotedev-server sqlite3`)
+`Usage Error: The 'yarn global' commands have been removed in 2.x - consider using 'yarn dlx' or a third-party plugin instead`
```
Comment (MajorLift) [inline README.md:60]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -49,7 +56,7 @@ To start the [React DevTools](https://github.com/facebook/react-devtools), run `
 
 To start the [Redux DevTools Extension](https://github.com/reduxjs/redux-devtools/tree/main/extension):
 
-- Install the package `remotedev-server` globally (e.g. `yarn global add remotedev-server`)
+- Install the package `remotedev-server` globally (e.g. `yarn dlx remotedev-server`)
```
Comment (MajorLift) [inline README.md:59]:
```suggestion
- Install the package `remotedev-server` globally (e.g. `yarn global add remotedev-server`)
```

File: README.md [hunk 3]
```diff
@@ -37,6 +42,8 @@ To get quick feedback from our shared code quality fitness functions before comm
 
 You can read more about them in our [testing documentation](./docs/testing.md#fitness-functions-measuring-progress-in-code-quality-and-preventing-regressions-using-custom-git-hooks).
 
+If you are using VS Code and are unable to make commits from the source control menu due to a "command not found" error, try these steps from the [Husky docs](https://typicode.github.io/husky/troubleshooting.html#command-not-found).
```
Comment (MajorLift) [inline README.md:45]:
```suggestion
If you are using VS Code and are unable to make commits from the source control sidebar due to a "command not found" error, try these steps from the [Husky docs](https://typicode.github.io/husky/troubleshooting.html#command-not-found).
```

**Review by mikesposito — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**